internal/rbd/nodeserver.go

```diff
@@ -215,8 +215,28 @@ func populateRbdVol(
 		rv.RbdImageName = imageAttributes.ImageName
 	}
 
+	err = rv.Connect(cr)
+	if err != nil {
+		log.ErrorLog(ctx, "failed to connect to volume %s: %v", rv, err)
+
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	// in case of any error call Destroy for cleanup.
+	defer func() {
+		if err != nil {
```

Comment:
On line 243 there is no `err`, but an error is returned never the less. `Destroy()` will not be called in that case. Better create an error instance there, log it, and return it.

Reply:
Yes, correct. update err instance should be fine now.